<p>
In recent research (see <a href="https://doi.org/10.1007/s10664-021-09950-y">Improving energy-efficiency by recommending Java collections</a>) it has been discovered that <code>ArrayList</code> is less energy-efficent than the implementations of Eclipse Collections or Apache Commons. In summary, using <code>ArrayList</code> may lead to higher Carbon footprint of your software.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
List<String> items = new ArrayList<>();
</pre>

<h2>Compliant Solution</h2>

<pre>
List<String> items = new org.eclipse.collections.impl.list.mutable.FastList();
</pre>
